feat: update README (#684)
openedx/frontend-app-authn: README.rst

|Build Status| |ci-badge| |Codecov| |license| |semantic-release| |npm_version|

frontend-app-authn
====================

Please tag **@openedx/vanguards** on any PRs or issues.  Thanks!

Introduction
------------

This is a micro-frontend application responsible for the login, registration and password reset functionality.

**What is the domain of this MFE?**

- Register page

- Login page

- Forgot password page

- Reset password page

> - Welcome page

Reviewer: Remove this.

     - ``https://support.example.com``

   * - ``ACTIVATION_EMAIL_SUPPORT_LINK``
     - The fully-qualified URL to the activation email support page in the target environment.
     - ``https://support.example.com``

   * - ``PASSWORD_RESET_SUPPORT_LINK``
     - The fully-qualified URL to the password reset support page in the target environment.
     - ``https://support.example.com``

   * - ``WELCOME_PAGE_SUPPORT_LINK``
     - The fully-qualified URL to the welcome support page in the target environment.
     - ``https://support.example.com``

   * - ``TOS_AND_HONOR_CODE``
     - The fully-qualified URL to the Honor code page in the target environment.
     - ``https://example.com/honor``

   * - ``TOS_LINK``
     - The fully-qualified URL to the Terms of service page in the target environment.
     - ``https://example.com/tos``

   * - ``PRIVACY_POLICY``
     - The fully-qualified URL to the Privacy policy page in the target environment.
     - ``https://example.com/privacy``

   * - ``INFO_EMAIL``
     - The valid email address for information query regarding the target environment.
     - ``info@example.com``

   * - ``ENABLE_DYNAMIC_REGISTRATION_FIELDS``

>      - Enables support for dynamic registration fields that helps in adding more fields in register form for getting more details about users.

Reviewer: ```suggestion
     - Enables support for configurable registration fields on the MFE. This flag must be enabled to show any registration field besides the default fields (`name`, `email`, `username`, `password`).
```

     - ``true`` | ``''`` (empty strings are falsy)

   * - ``DISABLE_ENTERPRISE_LOGIN``

>      - Enables support for disabling enterprise login.

Reviewer: nit:
```suggestion
     - Disables the enterprise login from Authn MFE.
```

     - ``true`` | ``''`` (empty strings are falsy)

edX-specific Environment Variables
**********************************

Furthermore, there are several edX-specific environment variables that enable integrations with closed-source services private to the edX organization, and might be unsupported in Open edX.

.. list-table:: edX-specific Environment Variables
   :widths: 30 50 20
   :header-rows: 1

   * - Name
     - Description / Usage
     - Example

   * - ``ENABLE_PROGRESSIVE_PROFILING``

>      - Enables support for progressive profiling that helps in redirecting to welcome page and getting more details about users.

Reviewer: ```suggestion
     - Enables support for progressive profiling. If enabled, users are redirected to a second page where data for optional registration fields can be collected. 
```

Reviewer: Now that I think about it, I think this should be moved to the above table. The progressive profiling page is needed for the optional fields and should be available to openedX users as well.
